src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -2352,7 +2352,7 @@ internal SqlFedAuthToken GetFedAuthToken(SqlFedAuthInfo fedAuthInfo)
                             }
                             else
                             {
-                                _fedAuthToken = authProvider.AcquireTokenAsync(authParamsBuilder).Result.ToSqlFedAuthToken();
+                                _fedAuthToken = Task.Run(async () => await authProvider.AcquireTokenAsync(authParamsBuilder)).Result.ToSqlFedAuthToken();
```

Comment:
Thanks @roji I did try that in the initial commit, but the driver denied to accept that change 🙈 

[Link to Failures](https://github.com/dotnet/SqlClient/runs/3344039854), primarily:
`typeof(System.FormatException): Index (zero based) must be greater than or equal to zero and less than the size of the argument list.`

On just reverting that to ".Result" (https://github.com/dotnet/SqlClient/pull/1213/commits/9adeb8cb5cdfcefa04de96b266ae413235153692) seems to work.

Reply:
But it's failing exactly because one of the tests is expecting `System.AggregateException` (which wraps the exception), but instead it actually throws `System.FormatException` (the one `System.AggregateException` wraps).

Reply:
One of the tests in question:

https://github.com/dotnet/SqlClient/blob/104e9a698cee44c5467bdf2cca457cba88c18a1d/src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/AADConnectionTest.cs#L159-L168

Reply:
It seems to be pointing to another issue we should handle internally - I'll try fixing it.

Reply:
Yeah, thanks @vonzshik. In general, AggregateException shouldn't be bubbling out of the driver, but should rather be unwrapped internally (e.g. via `GetAwait().GetResult()`)